src/main/java/homework10/FileLogger.java

```
@@ -0,0 +1,67 @@
+package homework10;
+
+import homework10.configuration.FileLoggerConfiguration;
+import homework10.enums.LoggingLevel;
+import homework10.exceptions.FileMaxSizeReachedException;
+import homework10.interfaces.Printable;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Random;
+
+public class FileLogger implements Printable {
+
+    private final FileLoggerConfiguration config;
+
+    public FileLogger(FileLoggerConfiguration config) {
+        this.config = config;
+    }
+
+    public boolean flag = false;
+
+    public void info(String message){
+        writeInFile(message, LoggingLevel.INFO);
+        flag = true;
+    }
+
+
+    public void debug(String message){
+        if (flag){
+        writeInFile(message, LoggingLevel.DEBUG);
+        } else {
+            throw new RuntimeException("DEBUG level started without INFO level");
+        }
+
+    }
+
+    private void writeToFile(File file, String message, LoggingLevel level, DateFormat timestamp) {
+        try (FileWriter writer = new FileWriter(file, true)) {
+            writer.write("[" + timestamp.format(new Date()) + "][" + level + "] " + "Message: " + message + "\n");
+        } catch (RuntimeException | IOException e) {
+            e.printStackTrace();

```

throw an exception instead of printing a stack trace